core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -143,10 +154,22 @@ abstract class AbstractIndex[K, V](@volatile var file: File, val baseOffset: Lon
    * @throws IOException if rename fails
    */
   def renameTo(f: File) {
-    try{ 
-      closeHandler()
-      Utils.atomicMoveWithFallback(file.toPath, f.toPath)
-    } 
+    try{
+      if(OperatingSystem.IS_WINDOWS) {
```

Comment:
Looked at the code.. Two problems here... Open and Close logic is different for FileChannel and Memory Map. Renaming memory map file is already in a abstract method. closing and opening a FileChannel can be put in a Util, but it will result in more changes. Return of value is small to make this happen.